go/processor/floria/run_context.go

```diff
@@ -26,42 +30,60 @@ type runContext struct {
 }
 
 func (r runContext) Call(kind tosca.CallKind, parameters tosca.CallParameters) (tosca.CallResult, error) {
-
 	if r.depth > MaxRecursiveDepth {
 		return tosca.CallResult{}, nil
 	}
 	r.depth++
 	defer func() { r.depth-- }()
 
-	snapshot := r.CreateSnapshot()
-	if err := transferValue(r, parameters.Value, parameters.Sender, parameters.Recipient); err != nil {
-		r.RestoreSnapshot(snapshot)
-		return tosca.CallResult{}, nil
-	}
-
 	codeHash := r.GetCodeHash(parameters.Recipient)
 	code := r.GetCode(parameters.Recipient)
 
-	sender := parameters.Sender
 	if kind == tosca.DelegateCall || kind == tosca.CallCode {
 		code = r.GetCode(parameters.CodeAddress)
 		codeHash = r.GetCodeHash(parameters.CodeAddress)
 	}
 
+	recipient := parameters.Recipient
+	var createdAddress tosca.Address
+	if kind == tosca.Create || kind == tosca.Create2 {
+		if parameters.Recipient == (tosca.Address{}) {
+			code = tosca.Code(parameters.Input)
+			codeHash = hashFromCode(code)
```

Comment:
I think in the interpreter implementations we accept `nil` pointer for code hashes in such cases. This should prevent the need to hash and cache codes that are only used once.

We can keep the hashing here now, it should not be a problem. But maybe, if we see performance issues with hashing of codes, this may be something to be removed.

Reply:
I will focus on correctness and keeping it as simple as possible first. We can also profile and tune the implementation when we are satisfied with the correctness.